### PR TITLE
Refactors Controllers to split Controllers and Routers

### PR DESCRIPTION
**Pull request by flovilmart**
- Improves tests and coverage, fix bugs

@nlutsenko that's what we were talking about yesterday.

### CODE REVIEW

**Comment by nlutsenko** [comment]
Looks good so far, will take a deeper look in a bit.

**Comment by flovilmart** [comment]
@nlutsenko, we should add a piece around architecture in the CONTRIBUTING.md

**Comment by flovilmart** [comment]
And could improve branch coverage in the controllers to 100% as the design begs for them to be fully test drive

File: spec/FilesController.spec.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+var FilesController = require('../src/Controllers/FilesController').FilesController;
+var Config = require("../src/Config");
+
+// Small additional tests to improve overall coverage
+describe("FilesController",()=>{
+  
+  it("should properly expand objects", (done) => {
```
Comment (nlutsenko) [inline spec/FilesController.spec.js]: Are we testing the fact that the controller doesn't expand non-necessary objects?
Can we have a reverse test? Say - expand a file with just a name, so it has the full url?

**Comment by nlutsenko** [comment]
Yup, that's why we have very good coverage on the SDKs - the controllers should be tested as much as possible... Agreed about architecture in CONTRIBUTING.md, but not sure how many people actually read it before sending a Pull Request, and in a little bit - that piece will actually disappear, since we won't need an architecture doc, as the code will look nicer :stuck_out_tongue_winking_eye: 

Can merge as is, the question about the test is not that relevant.

**Comment by flovilmart** [comment]
You can merge as is, as the coverage improves compared to the master. I'll make another PR for more tests if needed.

**Comment by nlutsenko** [comment]
We can make them as we go, merging now. Awesome work, sir!

File: spec/FilesController.spec.js (hunk 1 above)
Comment (flovilmart) [inline spec/FilesController.spec.js]: This is already covered by the integration tests that are run through Parse.File.save

**Pull request merged**